### ISSUE TITLE
[Property Annotations] Handle only component property annotations marked as @ComponentPropertyType

### ISSUE THREAD

**Issue by rinswind** [pkriens/cdi#17]
- **For**
    - Safe: otherwise must maintain a list of exceptions: `javax.inject`, `javax.enterprise.inject`, `org.osgi.service.cdi.annotatoin`, ....
- **Against**
    - It's from the DS API

**Comment by rotty3000** [comment]
No, because this means that none of the existing CDI annotations from the whole CDI echosystem could be used as component property types.

All we need to support is anything marked as Qualifier!

**Comment by rinswind** [comment]
The `@Qualifier` is used to bind dependency injection points to beans. A matter internal to the CDI container. If I am not mistaken want to also use qualifiers for something unrelated to dependency injection.

Conflating these two unrelated roles may cause issues:
1. We forbid component beans to be injected anywhere.
2. A user adds a qualifier to component bean to have a property
3. This can cause the component bean to become injectable somewhere.

I will can't make into definition errors all possible corner cases of this type. Rather it is better to have the user decide if they want their property annotations to double in a dependency injection role.

**Comment by rotty3000** [comment]
1. Qualifier in CDI is EXACTLY the same as "service properties" in OSGi. They are used to "target" more specifically what objects should be injected. So it's a perfect match.
2. There are tons of specifications based on CDI which implement @Qualifier, which we want to use as properties when we replace their concrete beans with OSGi services. The @Qualifer was exactly designed to be the joining force between different specifications and metadata to target what get's injected.


>    We forbid component beans to be injected anywhere.

I don't understand this argument.

>    A user adds a qualifier to component bean to have a property

Yeah, they already do this for that exact reason. It's natural CDI behavior.

>    This can cause the component bean to become injectable somewhere.

How so? Can you explain this assertion?

**Comment by rinswind** [comment]
**Suggestion**
- There is an internal blacklist: (internal)
          `javax.inject`, //@Named
          `javax.enterprise.inject`,
- Define an qualifier blacklist via the xml.
    - per class
- Support it with a meta-annotation for code we control

**Comment by rinswind** [comment]
Close with resolution: suggestion accepted

**Comment by rotty3000** [comment]
Reopen to indicate required action.

**Comment by rinswind** [comment]
Added a `ComponentPropertyIgnore` meta annotation.

Please review.

**Comment by rinswind** [comment]
I am still not sure how `beans.xml`, `osgi-beans.xml` and the runtime annotations relate to each other. Should I open an issue to clarify this?
